src/osd/OSD.cc

```diff
@@ -6901,6 +6901,10 @@ void OSD::_committed_osd_maps(epoch_t first, epoch_t last, MOSDMap *m)
         epoch_t bind_epoch = osdmap->get_epoch();
         service.set_epochs(NULL,&up_epoch, &bind_epoch);
 	do_restart = true;
+        int r;
+	set<int> avoid_ports;
+
+        start_waiting_for_healthy();
```

Comment:
i am not sure we need to optimise for rebind error at the expense of introducing more entropy into the code.